src/common/function/FunctionManager.cpp

```diff
@@ -1828,12 +1828,16 @@ FunctionManager::FunctionManager() {
     attr.maxArity_ = 1;
     attr.isAlwaysPure_ = true;
     attr.body_ = [](const auto &args) -> Value {
-      switch (args[0].get().type()) {
+      const Value &value = args[0].get();
+      switch (value.type()) {
         case Value::Type::NULLVALUE: {
           return Value::kNullValue;
         }
         case Value::Type::VERTEX: {
-          return args[0].get().getVertex().vid;
+          return value.getVertex().vid;
+        }
+        case Value::Type::EDGE: {
```

Comment:
Emm. We don't have concept name `id` of edge, better to avoid expose it to user.

Reply:
The function start with `_`.

Reply:
> The function start with `_`.

We can do that like add a "_id()" function, but I think this is kind of over worried, I will leave that to future improvement if actually needed.

Reply:
Maybe could add new function named `_edgeid`, emm.

Reply:
> Maybe could add new function named `_edgeid`, emm.

if we add the "_edgeid" function, we also need change the plan, the hashKey is id($-.e) need changed to _edgeid($-.e), special treatment for join on edge

```
| 12 | BiInnerJoin    | 18,21        |                | outputVar: {                                |
|    |                |              |                |   "colNames": [                             |
|    |                |              |                |     "v",                                    |
|    |                |              |                |     "e",                                    |
|    |                |              |                |     "e",                                    |
|    |                |              |                |     "u"                                     |
|    |                |              |                |   ],                                        |
|    |                |              |                |   "type": "DATASET",                        |
|    |                |              |                |   "name": "__BiInnerJoin_12"                |
|    |                |              |                | }                                           |
|    |                |              |                | inputVar: {                                 |
|    |                |              |                |   "rightVar": "__Project_11",               |
|    |                |              |                |   "leftVar": "__Filter_6"                   |
|    |                |              |                | }                                           |
|    |                |              |                | hashKeys: [                                 |
|    |                |              |                |   "id($-.e)"                                |
|    |                |              |                | ]                                           |
|    |                |              |                | probeKeys: [                                |
|    |                |              |                |   "id($-.e)"                                |
|    |                |              |                | ]                                           |
|    |                |              |                | kind: InnerJoin                             |
```

Reply:
check the new added “_joinkey” function @Shylock-Hg

Reply:
What happens if join on other types...